src/parser/expr.rs

```diff
@@ -72,48 +76,41 @@ impl<'s, 'a> Parser<'s, 'a> {
     //     | term
     fn sum(&mut self) -> ParseResult<Expr<'s, 'a>> {
         let mut left = self.term()?;
-        loop {
-            if self.accept_optional(tok![+])?.is_some() {
-                let right = self.term()?;
-                let span = self.spans.merge(&left, &right);
-                left = ExprKind::Add(self.alloc(left), self.alloc(right)).with_span(span);
-            } else if self.accept_optional(tok![-])?.is_some() {
-                let right = self.term()?;
-                let span = self.spans.merge(&left, &right);
-                left = ExprKind::Sub(self.alloc(left), self.alloc(right)).with_span(span);
-            } else {
-                return Ok(left);
-            }
+        while let Some((op, _span)) =
+            self.try_tokens([(tok![+], BinOp::Add), (tok![-], BinOp::Sub)])?
```

Comment:
I would rename this to something like accept_any or something like that. This name reminded me of something that accepts these tokens after one another, from the definition at least. That'd be a bit confusing since we don't do backtracking at this stage